core/state/snapshot/snapshot.go

```diff
@@ -503,6 +516,16 @@ func diffToDisk(bottom *diffLayer) *diskLayer {
 
 		snapshotFlushAccountItemMeter.Mark(1)
 		snapshotFlushAccountSizeMeter.Mark(int64(len(data)))
+
+		// Ensure we don't write too much data blindly. It's ok to flush, the
+		// root will go missing in case of a crash and we'll detect and regen
+		// the snapshot.
+		if batch.ValueSize() > ethdb.IdealBatchSize {
```

Comment:
Shouldn't we also flush the "full" batch when we try to write the storages?

Reply:
The important one is the delete-flush, since it may be unbounded (extremely large). The things we hold in memory are bounded by the memory limits of the difflayers.